src/braket/circuits/circuit.py

```diff
@@ -407,6 +416,42 @@ def _add_to_qubit_observable_set(self, result_type: ResultType) -> None:
         if isinstance(result_type, ObservableResultType) and result_type.target:
             self._qubit_observable_set.update(result_type.target)
 
+    def _check_if_qubit_measured(
+        self,
+        instruction: Instruction,
+        target: QubitSetInput | None = None,
+        target_mapping: dict[QubitInput, QubitInput] | None = None,
+    ) -> None:
+        """Checks if the target qubits are measured. If the qubit is already measured
+        the instruction will not be added to the Circuit.
+
+        Args:
+            instruction (Instruction): `Instruction` to add into `self`.
+            target (QubitSetInput | None): Target qubits for the
+                `instruction`. If a single qubit gate, an instruction is created for every index
+                in `target`.
+                Default = `None`.
+            target_mapping (dict[QubitInput, QubitInput] | None): A dictionary of
+                qubit mappings to apply to the `instruction.target`. Key is the qubit in
+                `instruction.target` and the value is what the key will be changed to.
+                Default = `None`.
+
+        Raises:
+            ValueError: If adding a gate or noise operation after a measure instruction.
+        """
+        if (
+            target
+            and target in self._measure_targets
+            or (target_mapping and all(targ in self._measure_targets for targ in target_mapping))
+            or (
+                instruction.target
+                and all(targ in self._measure_targets for targ in instruction.target)
```

Comment:
should this be all or any? Hopefully we're confident in the logic here through unit tests

Reply:
Based on the logic, this should be `any`. Updated it and updated a test to check for targets both in `self._measure_targets` and not.